tests/unit/dataactvalidator/test_a20_appropriations_1.py

```diff
@@ -0,0 +1,32 @@
+from tests.unit.dataactcore.factories.staging import AppropriationFactory
+from tests.unit.dataactcore.factories.domain import CGACFactory
+from tests.unit.dataactvalidator.utils import number_of_errors, query_columns
+from random import choice
+from string import ascii_uppercase, ascii_lowercase, digits
+
+_FILE = 'a20_appropriations_1'
+
+def test_column_headers(database):
+    expected_subset = {"row_number", "allocation_transfer_agency"}
+    actual = set(query_columns(_FILE, database))
+    assert expected_subset <= actual
+
+def test_success(database):
+    """ Test that agency codes are matched against cgac correctly """
+    approp = AppropriationFactory()
+    approp_null = AppropriationFactory(allocation_transfer_agency = None)
+    cgac = CGACFactory(cgac_code = approp.allocation_transfer_agency)
+
+    errors = number_of_errors(_FILE, database, models=[approp, approp_null, cgac])
+    assert errors == 0
+
+def test_failure(database):
+    """ Test a cgac not present in cgac table """
+    # These cgacs are different lengths to avoid being equal
```

Comment:
So the reason I've been leaning towards randomly generated data throughout these tests is that it eventually tests a broader range of possibilities.  If there are strange edge cases that occur for certain types of values, the random data gives us a better chance of finding those.  Does that make sense, or is the extra readability of the specific values worth dropping that?